druid/src/core.rs

```diff
@@ -1069,6 +1083,17 @@ impl WidgetState {
         self.request_update |= child_state.request_update;
         self.request_focus = child_state.request_focus.take().or(self.request_focus);
         self.timers.extend_drain(&mut child_state.timers);
+
+        let child_cursor = child_state.cursor.take();
+        if child_state.has_active || child_state.is_hot {
+            if let CursorSetting::Override(cursor) = &self.cursor_setting {
+                self.cursor = Some(Rc::clone(cursor));
+            } else if child_cursor.is_some() {
+                self.cursor = child_cursor;
+            } else if let CursorSetting::Set(cursor) = &child_state.cursor_setting {
+                self.cursor = Some(Rc::clone(cursor));
+            }
```

Comment:
this logic is hurting my head a bit.

Is there a situation where the `child_cursor.is_some()` ends up unintentionally overriding the child's `cursor_setting`?

or... what if both a parent and a child have called `set_cursor`, and the mouse moves over to the child, and then back to the parent, does the parent end up correctly resetting the cursor? This happens at the grandparent, in its own `merge_up`?

I think I'm satisfied, but want to at least write that concern down.

Reply:
I've re-written the logic so that it's only merging `self.cursor_change` and `child.cursor` to make `self.cursor`. (i.e. `child.cursor_change` is no longer involved, so there's less multi-generational thinking involved)

I've also tried to make it clearer that `cursor` gets reset on every pass through the tree, whereas `cursor_change` is persistent.